[docs] current master fails in docs test

```
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 20 source files that are out of date
updating environment: [new config] 37 added, 0 changed, 0 removed
reading sources... [  5%] C-API                                                 
Exception occurred:
  File "/home/travis/.local/lib/python3.7/site-packages/sphinx/domains/c.py", line 3053, in object_type
    raise NotImplementedError()
NotImplementedError
The full traceback has been saved in /tmp/sphinx-err-z9o6pn64.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:20: recipe for target 'html' failed
make: *** [html] Error 2
The command "bash .ci/test.sh" exited with 255.
```
The reason is in Sphinx 3 release.

Last successful test:

```
Collecting sphinx
  Downloading Sphinx-2.4.4-py3-none-any.whl (2.7 MB)
     |████████████████████████████████| 2.7 MB 4.2 MB/s 
```

In failed test:

```
Collecting sphinx
  Downloading Sphinx-3.0.0-py3-none-any.whl (2.8 MB)
     |████████████████████████████████| 2.8 MB 3.8 MB/s 
```


I suppose the root cause is in `breathe` and `sphinx` incompatibility, like it was in https://github.com/michaeljones/breathe/issues/465, https://github.com/sphinx-doc/sphinx/issues/7085.

--- a/.ci/test_r_package.sh
+++ b/.ci/test_r_package.sh
@@ -38,10 +38,8 @@ if [[ $OS_NAME == "macos" ]]; then
     brew install qpdf
     brew cask install basictex
     export PATH="/Library/TeX/texbin:$PATH"
-
-    # Workaround for "/Library/TeX/texbin/tlmgr: unexpected return value from verify_checksum: -5"
-    # https://tug.org/pipermail/tex-live/2020-February/044772.html
-    curl -fsSL https://www.preining.info/rsa.asc | sudo tlmgr key add -
+    sudo tlmgr update --self
+    sudo tlmgr install inconsolata helvetic
 
     wget -q https://cran.r-project.org/bin/macosx/R-${R_MAC_VERSION}.pkg -O R.pkg
     sudo installer \
@@ -58,10 +56,6 @@ if [[ $OS_NAME == "macos" ]]; then
             /Library/Frameworks/R.framework/Versions/${R_MAJOR_MINOR}/Resources/lib/libomp.dylib
     fi
 fi
-
-# Fix "! LaTeX Error: File `inconsolata.sty' not found."
-sudo tlmgr update --self
-sudo tlmgr install inconsolata helvetic
 
 conda install \
     -y \

--- a/.ci/test_r_package.sh
+++ b/.ci/test_r_package.sh
@@ -38,8 +38,8 @@ if [[ $OS_NAME == "macos" ]]; then
     brew install qpdf
     brew cask install basictex
     export PATH="/Library/TeX/texbin:$PATH"
-    sudo tlmgr update --self
-    sudo tlmgr install inconsolata helvetic
+    sudo tlmgr --verify-repo=none update --self
+    sudo tlmgr --verify-repo=none install inconsolata helvetic
 
     wget -q https://cran.r-project.org/bin/macosx/R-${R_MAC_VERSION}.pkg -O R.pkg
     sudo installer \

--- a/.ci/test_r_package.sh
+++ b/.ci/test_r_package.sh
@@ -38,8 +38,10 @@ if [[ $OS_NAME == "macos" ]]; then
     brew install qpdf
     brew cask install basictex
     export PATH="/Library/TeX/texbin:$PATH"
-    sudo tlmgr update --self
-    sudo tlmgr install inconsolata helvetic
+
+    # Workaround for "/Library/TeX/texbin/tlmgr: unexpected return value from verify_checksum: -5"
+    # https://tug.org/pipermail/tex-live/2020-February/044772.html
+    curl -fsSL https://www.preining.info/rsa.asc | sudo tlmgr key add -
 
     wget -q https://cran.r-project.org/bin/macosx/R-${R_MAC_VERSION}.pkg -O R.pkg
     sudo installer \
@@ -56,6 +58,10 @@ if [[ $OS_NAME == "macos" ]]; then
             /Library/Frameworks/R.framework/Versions/${R_MAJOR_MINOR}/Resources/lib/libomp.dylib
     fi
 fi
+
+# Fix "! LaTeX Error: File `inconsolata.sty' not found."
+sudo tlmgr update --self
+sudo tlmgr install inconsolata helvetic
 
 conda install \
     -y \

--- a/docs/requirements_base.txt
+++ b/docs/requirements_base.txt
@@ -1,3 +1,3 @@
-sphinx
+sphinx < 3.0
 sphinx_rtd_theme >= 0.3
 mock; python_version < '3'

